src/core/toast.ts

```diff
@@ -63,28 +63,32 @@ toast.promise = <T>(
   promise: Promise<T>,
   msgs: {
     loading: Renderable;
-    success: ValueOrFunction<Renderable, T>;
-    error: ValueOrFunction<Renderable, any>;
+    success?: ValueOrFunction<Renderable, T>;
+    error?: ValueOrFunction<Renderable, any>;
   },
   opts?: DefaultToastOptions
 ) => {
   const id = toast.loading(msgs.loading, { ...opts, ...opts?.loading });
 
   promise
     .then((p) => {
-      toast.success(resolveValue(msgs.success, p), {
-        id,
-        ...opts,
-        ...opts?.success,
-      });
+      if (msgs.success)
+        toast.success(resolveValue(msgs.success, p), {
```

Comment:
The existing `ValueOrFunction` allows for null as a return value, it also seems reasonable that if a user returns null from the success/error function it could skip the toast.  This seems preferable since especially in the case of errors, whether to show an error may itself be conditional on the type of Error raised.

Reply:
Agree! Added this behaviour as well.